File: src/Bing.Scheduler/Core/Result.cs
namespace Bing.Scheduler.Core
{
    /// <summary>
    /// 结果
    /// </summary>
    public class Result
    {
        /// <summary>
        /// 状态码
        /// </summary>
        public int Code { get; set; }

        /// <summary>
        /// 是否成功
        /// </summary>
        public bool Success { get; set; }

        /// <summary>
        /// 消息
        /// </summary>
        public string Message { get; set; }

        /// <summary>
        /// 数据
        /// </summary>
        public dynamic Data { get; set; }
    }
}

File: samples/Bing.Demo/ui/src/app/shared/bing/models/result.ts
// ============== 服务端结果约定格式 ===============
// Copyright 2019 简玄冰
// Licensed under the MIT license
// ================================================
import { HttpErrorResponse } from '@angular/common/http';

/**
 * 服务端返回的标准结果
 */
export class Result<T> {
    /**
     * 状态码
     */
    code: StateCode;
    /**
     * 消息
     */
    message: string;
    /**
     * 数据
     */
    data: T;
}

/**
 * 错误结果
 */
export class FailResult {
    /**
     * 状态码
     */
    code: StateCode;
    /**
     * 消息
     */
    message: string;

    /**
     * 初始化错误结果
     * @param result 结果
     * @param errorResponse Http响应错误
     */
    constructor(result?: Result<any>, public errorResponse?: HttpErrorResponse) {
        if (result) {
            this.code = result.code;
            this.message = result.message;
        }
    }
}

/**
 * 状态码
 */
export enum StateCode {
    /**
     * 成功
     */
    Ok = 1,
    /**
     * 失败
     */
    Fail = 2
}
